src/local_sensitivity/backsolve_adjoint.jl

```diff
@@ -81,6 +82,76 @@ end
   return ODEProblem(odefun,z0,tspan,p,callback=cb)
 end
 
+
+
+@noinline function SDEAdjointProblem(sol,sensealg::BacksolveAdjoint,
+                                     g,t=nothing,dg=nothing;
+                                     checkpoints=sol.t,
+                                     callback=CallbackSet(),kwargs...)
+  @unpack f, p, u0, tspan = sol.prob
+  tspan = reverse(tspan)
+  discrete = t != nothing
+
+  p === DiffEqBase.NullParameters() && error("Your model does not have parameters, and thus it is impossible to calculate the derivative of the solution with respect to the parameters. Your model must have parameters to use parameter sensitivity calculations!")
+  numstates = length(u0)
+  numparams = length(p)
+
+  len = length(u0)+numparams
+  λ = similar(u0, len)
+
+  drift_function = ODEFunction(sol.prob.f)
+  sense_drift = ODEBacksolveSensitivityFunction(g,sensealg,discrete,sol,dg,drift_function,drift_function.colorvec)
+
+  diffusion_function = ODEFunction(sol.prob.g)
+  sense_diffusion = ODEBacksolveSensitivityFunction(g,sensealg,discrete,sol,dg,diffusion_function,diffusion_function.colorvec)
+
+  init_cb = t !== nothing && tspan[1] == t[end]
+  cb = generate_callbacks(sense_drift, g, λ, t, callback, init_cb)
+  checkpoints = ischeckpointing(sensealg, sol) ? checkpoints : nothing
+  if checkpoints !== nothing
+    cb = backsolve_checkpoint_callbacks(sense_drift, sol, checkpoints, cb)
+  end
+
+  z0 = [vec(zero(λ)); vec(sense_drift.y)]
+  #@show z0, vec(zero(λ)), sense_drift.y
+
+  original_mm = sol.prob.f.mass_matrix
+  if original_mm === I
+    mm = I
+  else
+    sense_drift.diffcache.issemiexplicitdae && @warn "`BacksolveAdjoint` is likely to fail on semi-explicit DAEs, if memory is a concern, please consider using InterpolatingAdjoint(checkpoint=true) instead."
+    len2 = length(z0)
+    mm = zeros(len2, len2)
+    idx = 1:numstates
+    copyto!(@view(mm[idx, idx]), sol.prob.f.mass_matrix')
+    idx = numstates+1:numstates+1+numparams
+    copyto!(@view(mm[idx, idx]), I)
+    idx = len+1:len2
+    copyto!(@view(mm[idx, idx]), sol.prob.f.mass_matrix)
+  end
+
+  sdefun = SDEFunction(sense_drift,sense_diffusion,mass_matrix=mm)
+
+  # replicated noise
+  _sol = deepcopy(sol)
+  if typeof(_sol.W.dW) <: Number
+    noisearray =  _sol.W.W
+  elseif StochasticDiffEq.is_diagonal_noise(sol.prob) && length(sol.W.dW)==1
+    noisearray = [W[1] for W in _sol.W.W]
```

Comment:
you shouldn't have to do that?

Reply:
You should be able to just reverse the array of array IIUC. If not, we need to just create a new dispatch to reverse.

Reply:
the reverse was actually not the issue. I could construct the NoiseGrid and the SDEProblem but when calling solve on the inplace function I got a broadcasting error in perform_step!

Reply:
This will only work on length 1 SDEs though, which is somewhat odd. This is the next thing to fix.